src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java

```diff
@@ -146,17 +145,16 @@ public boolean isBalanceDiffConsistent(Set<Hash> approvedHashes, Map<Hash, Long>
             countedTx.add(solidEntryPointHash);
         });
 
-        final Queue<Hash> nonAnalyzedTransactions = new LinkedList<>(Collections.singleton(milestoneHash));
+        final Queue<Hash> nonAnalyzedTransactions = new LinkedList<>(Collections.singleton(transactionHash));
         Hash transactionPointer;
         while ((transactionPointer = nonAnalyzedTransactions.poll()) != null) {
             if (visitedTransactions.add(transactionPointer)) {
-
                 try {
                     final TransactionViewModel transactionViewModel = TransactionViewModel.fromHash(tangle,
                             transactionPointer);
-
-                    if (milestoneService.transactionBelongsToMilestone(transactionViewModel, milestoneIndex)) {
-
+                    // only take transactions into account that have not been confirmed, yet (that are not included in
+                    // the ledger state)
+                    if (!milestoneService.isTransactionConfirmed(transactionViewModel)) {
```

Comment:
I think we overlooked something :-(
This code is different than before.

Before you were fed from the parameters a `milestoneIndex` that could never change.
Now it can change in the background...

Reply:
it was always called with snapshotProvider.getLatestSnapshot().getIndex() because thats what the method was for - generate the stateDiff compared to the confirmed part of the tangle.

We now fix this value in the little wrapper

    @Override
    public boolean isTransactionConfirmed(TransactionViewModel transaction) {
        return isTransactionConfirmed(transaction, snapshotProvider.getLatestSnapshot().getIndex());
    }

It doesnt matter if the wrapper fixes the value or the caller. The underlying isTransactionConfirmed(TransactionViewModel, milestoneIndex) will work with a fixed value that can not magically change anymore

Reply:
Now I got you -> fixed 💃